pkg/validation/validation.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"strings"
+
+	nropv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+)
+
+const (
+	// ValidationNodeGroupsError specifies the condition reason when node groups failed to pass validation
+	ValidationNodeGroupsError = "ValidationErrorUnderNodeGroups"
+)
+
+// MachineConfigPoolDuplicates selected MCPs for duplicates
+// TODO: move it under the validation webhook once we will have one
+func MachineConfigPoolDuplicates(mcps []*machineconfigv1.MachineConfigPool) error {
+	duplicates := map[string]int{}
+	for _, mcp := range mcps {
+		if _, ok := duplicates[mcp.Name]; !ok {
+			duplicates[mcp.Name] = 0
+		}
+		duplicates[mcp.Name] += 1
+	}
+
+	var duplicateErrors []string
+	for mcpName, count := range duplicates {
+		if count > 1 {
+			duplicateErrors = append(duplicateErrors, fmt.Sprintf("the MachineConfigPool %q selected by at least two node groups", mcpName))
+		}
+	}
+
+	if len(duplicateErrors) > 0 {
+		return fmt.Errorf(strings.Join(duplicateErrors, "; "))
+	}
+
+	return nil
+}
+
+// NodeGroups validates the node groups for nil values and duplicates.
+// TODO: move it under the validation webhook once we will have one
+func NodeGroups(nodeGroups []nropv1alpha1.NodeGroup) error {
+	if err := nodeGroupsMachineConfigPoolSelector(nodeGroups); err != nil {
+		return err
+	}
+
+	if err := nodeGroupsDuplicates(nodeGroups); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO: move it under the validation webhook once we will have one
+func nodeGroupsMachineConfigPoolSelector(nodeGroups []nropv1alpha1.NodeGroup) error {
+	for _, nodeGroup := range nodeGroups {
+		if nodeGroup.MachineConfigPoolSelector == nil {
+			return fmt.Errorf("one of the node groups does not have machineConfigPoolSelector")
+		}
+	}
+
+	return nil
+}
+
+// TODO: move it under the validation webhook once we will have one
```

Comment:
I'm not sure we want to invest into a validation webhook considering the purpose of this operator, but i'm open to such opportunity if there is need and we have capacity

Reply:
in general, it is not too hard and provides us some guarantees regarding to API values
Anyway we can discuss it in the future

Reply:
I'm open to that in general, let's just prioritize properly and not overinvest if we can help it.